apis/v1alpha2/validation/gateway.go

```diff
@@ -44,7 +46,32 @@ func validateGatewaySpec(spec *gatewayv1a2.GatewaySpec, path *field.Path) field.
 // to the Gateway API specification.
 func validateGatewayListeners(listeners []gatewayv1a2.Listener, path *field.Path) field.ErrorList {
 	// TODO [danehans]: Add additional validation of listener fields.
-	return validateListenerHostname(listeners, path)
+	var errs field.ErrorList
+	e1:=validateListenerTLSConfig(listeners, path)
+	errs = append(errs,e1...)
+	e2:=validateListenerHostname(listeners, path)
+	errs = append(errs,e2...)
+	return errs
+}
+
+func validateListenerTLSConfig(listeners []gatewayv1a2.Listener,path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	tlsHttpsProtocols := []gatewayv1a2.ProtocolType{gatewayv1a2.HTTPSProtocolType,gatewayv1a2.TLSProtocolType}
+	for i,l := range listeners{
+		if !contains(tlsHttpsProtocols,l.Protocol) && l.TLS != nil {
+			errs = append(errs,field.Forbidden(path.Index(i).Child("tls"),fmt.Sprintf("should be empty for protocol %v",l.Protocol)))
+		}
+	}
+	return errs
+}
+
+func contains(protocolTypes []gatewayv1a2.ProtocolType, protocolType gatewayv1a2.ProtocolType)  bool {
+	for _, v := range protocolTypes {
+		if reflect.DeepEqual(v,protocolType) {
```

Comment:
Why do we need DeepEqual() here?

Reply:
removed the DeepEqual apparently the ProtocolType is a simple wrapper over string